chore[gen-2]:ENG-8051 added model-name and api-key in the builder.sdkInfo-event (#3878)
BuilderIO/builder: .changeset/fifty-apples-walk.md

---
"@builder.io/sdk-angular": major
"@builder.io/sdk-react-nextjs": major
"@builder.io/sdk-qwik": major
"@builder.io/sdk-react": major
"@builder.io/sdk-react-native": major
"@builder.io/sdk-solid": major
"@builder.io/sdk-svelte": major

> "@builder.io/sdk-vue": major

Reviewer: one small caveat: Qwik, NextJS and Angular SDKs are still in beta `v0.x`, so for those we will mark this as a `minor` update instead of `major` to avoid them becoming `v1.0`

---

- BREAKING CHANGE :firecracker: : updated subscribeToEditor arguments:
from:
    subscribeToEditor('page', () => { ... }, options: {trustedHosts:['...']})
to:
    subscribeToEditor({
        apiKey: '...',
        model: '...',
        trustedHosts: ['...'],
        callback: () => { ... }

>     })

Reviewer: Can you format this so the 🧨 appears properly and the code is in code blocks? 

Also, I'd explicitly state the two changes, which are that a) arguments are now passed as named parameters in object argument, and that b) `apiKey` is now a required field

---
"@builder.io/sdk-angular": minor
"@builder.io/sdk-react-nextjs": minor
"@builder.io/sdk-qwik": minor
"@builder.io/sdk-react": major
"@builder.io/sdk-react-native": major
"@builder.io/sdk-solid": major
"@builder.io/sdk-svelte": major
"@builder.io/sdk-vue": major
---

> - BREAKING CHANGE 🧨 : updated subscribeToEditor arguments:
>     - arguments are now passed as an named argument object
>     - `apiKey` is now required field
> 
>     Argument change -
>     from:
>         subscribeToEditor('page', () => { ... }, options: {trustedHosts:['...']})
>     to:
>         subscribeToEditor({
>             apiKey: '...',
>             model: '...',
>             trustedHosts: ['...'],
>             callback: () => { ... }
>         })

Reviewer: ```suggestion
- BREAKING CHANGE 🧨 : updated `subscribeToEditor` arguments:
    - arguments are now passed as a named argument object
    - `apiKey` is now a required field

Example:
    - from:
    ```ts
        subscribeToEditor('page', () => { ... }, options: {trustedHosts:['...']})
    ```
    - to:
    ```ts
        subscribeToEditor({
            apiKey: '...',
            model: '...',
            trustedHosts: ['...'],
            callback: () => { ... }
        })
    ```
```

---

> "@builder.io/sdk": patch

Reviewer: `@builder.io/sdk` is the package containing core logic of gen1 SDK. What you want is to make a changeset for every SDK with the name `@builder.io/sdk-*`

---

> chore:added modal-name and api-key in sdkInfo-event

Reviewer: ```suggestion
chore: send `apiKey` and `model` to Visual Editor to improve editing experience.
```
